src/Catalyst.Core.Modules.Dfs.Tests/CoreApi/BitswapApiTest.cs

```diff
@@ -0,0 +1,433 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.Dfs.BlockExchange.Protocols;
+using Catalyst.Abstractions.Options;
+using Catalyst.Core.Modules.Dfs.BlockExchange;
+using Catalyst.Core.Modules.Dfs.BlockExchange.Protocols;
+using FluentAssertions;
+using Lib.P2P;
+using MultiFormats;
+using Xunit;
+
+namespace Catalyst.Core.Modules.Dfs.Tests.CoreApi
+{
+    public class BitswapApiTest
+    {
+        IDfs ipfs = TestFixture.Ipfs;
+        IDfs ipfsOther = TestFixture.IpfsOther;
+
+        [Fact]
+        public async Task Wants()
+        {
+            await ipfs.StartAsync();
+            try
+            {
+                var cts = new CancellationTokenSource();
+                var block = new DagNode(Encoding.UTF8.GetBytes("BitswapApiTest unknown block"));
+                Task wantTask = ipfs.Bitswap.GetAsync(block.Id, cts.Token);
+
+                var endTime = DateTime.Now.AddSeconds(10);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                        throw new Xunit.Sdk.XunitException("wanted block is missing");
+                    await Task.Delay(100);
+                    var w = await ipfs.Bitswap.WantsAsync();
+                    if (w.Contains(block.Id))
+                        break;
+                }
+
+                cts.Cancel();
+                var wants = await ipfs.Bitswap.WantsAsync();
+                wants.ToArray().Should().NotContain(block.Id);
+                Assert.True(wantTask.IsCanceled);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+            }
+        }
+
+        [Fact]
+        public async Task Unwant()
+        {
+            await ipfs.StartAsync();
+            try
+            {
+                var block = new DagNode(Encoding.UTF8.GetBytes("BitswapApiTest unknown block 2"));
+                Task wantTask = ipfs.Bitswap.GetAsync(block.Id);
+
+                var endTime = DateTime.Now.AddSeconds(10);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                        throw new Xunit.Sdk.XunitException("wanted block is missing");
+
+                    // Assert.Fail("wanted block is missing");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)